sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/command.py

```diff
@@ -5,31 +5,32 @@
 
 from azure.ai.ml._internal._schema.node import InternalBaseNodeSchema, NodeType
 from azure.ai.ml._schema import NestedField, StringTransformedEnum
-from azure.ai.ml._schema.component.retry_settings import RetrySettingsSchema
-from azure.ai.ml._schema.core.fields import DistributionField
+from azure.ai.ml._schema.job import ParameterizedParallelSchema, ParameterizedCommandSchema
 from azure.ai.ml._schema.job.job_limits import CommandJobLimitsSchema
-from azure.ai.ml._schema.job_resource_configuration import JobResourceConfigurationSchema
 
 
-class CommandSchema(InternalBaseNodeSchema):
+class CommandSchema(InternalBaseNodeSchema, ParameterizedCommandSchema):
+    class Meta:
+        exclude = ["code", "distribution"]  # internal command doesn't have code & distribution
+
     type = StringTransformedEnum(allowed_values=[NodeType.COMMAND], casing_transform=lambda x: x)
-    compute = fields.Str()
+    # do we need to overwrite this?
     environment = fields.Str()
```

Comment:
Also wondering why overwrite environment to str

Reply:
We just skip many checks for internal component and leave it to server-side to avoid blocking 1P users. But let me re-enable this and see what will happen given it looks weird